components/leaderboard/CountdownTimer.tsx

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from 'react'
+import { Box } from 'components/OffsetBorder/Box'
+import { intervalToDuration, nextMonday, set } from 'date-fns'
+import type { Duration } from 'date-fns'
+
+const customFormatDuration = (x: Duration) =>
+  `${x.days ? x.days + 'd : ' : ''}${x.hours ? x.hours + 'hr : ' : ''}${
+    x.minutes ? x.minutes + 'min : ' : ''
+  }${x.seconds ? x.seconds + 'sec' : '0sec'}`
+
+const CountdownTimer = () => {
+  const now = new Date()
+  const [$time, $setTime] = useState(now)
+  const end = nextMonday(
+    set(now, { hours: 0, minutes: 0, seconds: 0, milliseconds: 0 })
+  )
+  useEffect(() => {
+    const i = setInterval(() => $setTime(new Date()), 3e3)
```

Comment:
This looks really laggy with 3000 ms, it should be a lot lower. Ideally it should use requestAnimationFrame but setInterval should be fine since we only show 1 second intervals and we don't need less than that.

Also why is this using scientific notation? Can we just use normal numbers that people are used to?

```suggestion
    const i = setInterval(() => $setTime(new Date()), 1000)
```

Reply:
I find scientific notation very scannable. I've adjusted to 1ms cycles